src/gui/controllers/ManualControlModuleController.java

```diff
@@ -0,0 +1,339 @@
+            return;
+
+        switch (keyEvent.getText()) {
+            case "w":
+                wDown = false;
+                break;
+            case "s":
+                sDown = false;
+                break;
+            case "a":
+                aDown = false;
+                break;
+            case "d":
+                dDown = false;
+                break;
+            case "e":
+                eDown = false;
+                break;
+        }
+    }
+
+    /**
+     * Handle click in the Full Auto Mode-button. Add Full Mode request and pulls the server.
+     * @param mouseEvent Triggering event
+     */
+    public void handleFullModeButtonClick(MouseEvent mouseEvent) {
+        if (mouseEvent.getButton() != MouseButton.PRIMARY)
+            return;
+        Server.getInstance().getRequestBuilder().addSetModeRequest(ControlMode.FULL_AUTO);
+        Server.getInstance().pull();
+    }
+
+    /**
+     * Handle click in the Half Auto Mode-button. Add Half Mode request and pulls the server.
+     * @param mouseEvent Triggering event
+     */
+    public void handlHalfModeButtonClick(MouseEvent mouseEvent) {
+        if (mouseEvent.getButton() != MouseButton.PRIMARY)
+            return;
+        Server.getInstance().getRequestBuilder().addSetModeRequest(ControlMode.SEMI_AUTO);
+        Server.getInstance().pull();
+    }
+
+    /**
+     * Handle click in the Manual Mode-button. Add Manual Mode request and pulls the server.
+     * @param mouseEvent Triggering event
+     */
+    public void handleManualModeButtonClick(MouseEvent mouseEvent) {
+        if (mouseEvent.getButton() != MouseButton.PRIMARY)
+            return;
+        Server.getInstance().getRequestBuilder().addSetModeRequest(ControlMode.MANUAL);
+        Server.getInstance().pull();
+    }
+
+    /**
+     * WASD Control method. This mehtod is meant to be run on a separate thread to continually controll the car when the
+     * WASD state is true (tracking). Pulls the server every 100ms with updated speed and turn values.
+     */
+    synchronized private void wasdControl() {
+        while(wasdState.get()) {
+            if ((wDown && sDown) || (!wDown && !sDown)) {
+                speed = 0;
+            }
+            else if (wDown) {
+                speed += KEY_SPEED_STEP;
+            }
+            else if (sDown) {
+                speed -= KEY_SPEED_STEP;
+            }
+
+
+            if ((aDown && dDown) || (!aDown && !dDown)) {
+                turn = 0;
+            }
+            else if (aDown) {
+                turn = -(MAX_TURN - 1);
+            }
+            else if (dDown) {
+                turn = MAX_TURN - 1;
+            }
+
+            if (eDown) {
+                turn = 0;
+                speed = 0;
+                Server.getInstance().getRequestBuilder().addEmergencyStopRequest();
+            }
+
+            if (speed > MAX_SPEED)
+                speed = MAX_SPEED - 1;
+
+            if(speed < -MAX_SPEED)

```

Comment:
Min speed is -128 = -0xb0

Reply:
This will be fixed with changes to communication protocol. Current implementation has the implemented limits.